modules/base/dashboard/dashboarditemtimevaryingtext.cpp

```diff
@@ -0,0 +1,161 @@
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2025                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/dashboard/dashboarditemtimevaryingtext.h>
+
+#include <fstream>
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/util/timemanager.h>
+#include <openspace/json.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/logging/logmanager.h>
+
+
+
+namespace {
+    constexpr std::string_view _loggerCat = "DashboardItemTimeVaryingText";
+
+    constexpr openspace::properties::Property::PropertyInfo TextBeforeInfo = {
+        "TextBefore",
+        "Text Before",
+        "Optional text before the time varying text to be displayed.",
+        openspace::properties::Property::Visibility::User
+    };
+    constexpr openspace::properties::Property::PropertyInfo DataFileInfo = {
+        "DataFile",
+        "Data File Path",
+        "The file path to the JSON data.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    struct [[codegen::Dictionary(DashboardItemTimeVaryingText)]] Parameters {
```

Comment:
```suggestion
    // This `DashboardItem` displays text based on the content of a provided data file. The
    // value that is displayed depends on the current in-game simulation time.
    struct [[codegen::Dictionary(DashboardItemTimeVaryingText)]] Parameters {
```

Should add some more description about the file format once that is settled (see below)